cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -352,32 +359,38 @@ Future<Void> clusterOperatorSecret(Supplier<Date> dateSupplier) {
      * due to a new CA key. It is not necessary when the CA certificate is replace while retaining the existing key.
      */
     Future<Void> rollingUpdateForNewCaKey() {
-        List<String> reason = new ArrayList<>(2);
+        RestartReasons kafkaPodRollReasons = RestartReasons.empty();
 
         if (clusterCa.keyReplaced()) {
-            reason.add("trust new cluster CA certificate signed by new key");
+            kafkaPodRollReasons.add(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED);
         }
 
         if (clientsCa.keyReplaced()) {
-            reason.add("trust new clients CA certificate signed by new key");
+            kafkaPodRollReasons.add(RestartReason.CLIENT_CA_CERT_KEY_REPLACED);
         }
 
-        if (!reason.isEmpty()) {
+        if (clusterCa.keyReplaced() || kafkaPodRollReasons.shouldRestart()) {
```

Comment:
This is an example of the hacky approach mentioned above. Originally, this had a single condition which applied to both. Not you have two conditions joined by OR. But one of them is for ZooKeeper only and one of them is for Kafka only.

Reply:
:+1: